draco/analysis/flagging.py

```diff
@@ -702,6 +702,87 @@ def process(self, stream):
         return out
 
 
+class ThresholdVisWeightTime(task.SingleTask):
+    """Create a mask to set low weights to zero in a single frequency/time sample.
+
+    This task identifies individual baseline/frequency/time samples that have weights
+    below a specified threshold relative to the median time at a given
+    frequency and baseline. The user can also specify an absolute threshold. A mask
+    is then constructed that masks all baselines at each offending frequency/time
+    sample.
+
+    Parameters
+    ----------
+    relative_threshold : float, optional
+        Relative threshold for masking. Default: 1e-6.
+    absolute_threshold : float, optional
+        Any weights with values less than this number will be set to zero. Default: None.
+    """
+
+    relative_threshold = config.Property(proptype=float, default=1e-6)
+    absolute_threshold = config.Property(proptype=float, default=None)
+
+    def process(self, stream):
+        """Make a mask based on low weights.
+
+        Parameters
+        ----------
+        stream : `.core.container` with `weight` attribute
+            Input timestream or sidereal stream.
+
+        Returns
+        -------
+        out : `.core.container.RFIMask`
+            RFIMask container with mask (set to True for flagged samples).
+        """
+
+        # Find time axis and make the output container depending on what 'stream' is
+        if "ra" in stream.axes:
+            out = containers.SiderealRFIMask(axes_from=stream, attrs_from=stream)
+        elif "time" in stream.axes:
+            out = containers.RFIMask(axes_from=stream, attrs_from=stream)
+        else:
+            raise TypeError(f"Require Timestream or SiderealStream. Got {type(stream)}")
+
+        # Ensure stream is distributed in frequency
+        stream.redistribute("freq")
+
+        # Get weights
+        weight = stream.weight[:]
+        local_weight = weight.local_array
+
+        # Compute median over time for each frequency and baseline
+        time_median = np.median(local_weight, axis=2)
+
+        # Compute mask that is True if any baseline at a given frequency/time
+        # has a small but nonzero weight relative to the median over time
+        mask = np.any(
+            (local_weight < self.relative_threshold * time_median[:, :, np.newaxis])
+            & (local_weight != 0),
+            axis=1,
+        )
+
+        # If absolute threshold is also specified, include this in mask
+        if self.absolute_threshold is not None:
+            mask = np.any(
+                (local_weight < self.absolute_threshold) & (mask[:, np.newaxis, :]),
```

Comment:
won't this only mask times/frequencies that have already been masked by the relative threshold?

Reply:
Agreed. How about this?
```
if self.absolute_threshold is not None:
    mask = np.any(
        ((local_weight < self.absolute_threshold) & (local_weight != 0)) | mask[:, np.newaxis, :],
        axis=1,
    )
```

Reply:
If I understand correctly, I think you'd want to combine the masks with an `|` after the `np.any`, and maybe also exclude zero weights from the comparison, as done above.